backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py

```diff
@@ -259,11 +261,17 @@ def get_s3_consumption_data(uri):
             }
 
     @staticmethod
-    def list_shared_databases_tables_with_env_group(environmentUri: str, groupUri: str):
+    @ResourcePolicyService.has_resource_permission(LIST_ENVIRONMENT_DATASETS)
+    def list_shared_databases_tables_with_env_group(uri: str, group_uri: str):
         context = get_context()
+        if group_uri not in context.groups:
+            raise exceptions.UnauthorizedOperation(
+                action='LIST_ENVIRONMENT_GROUP_DATASETS',
+                message=f'User: {context.username} is not a member of the team {group_uri}',
```

Comment:
nit: this is 1 example where we may not want to use input in output error message in the case of bad inputs being used to reflect back